src/ghostfat.c

```diff
@@ -38,6 +38,12 @@
 
 #define STATIC_ASSERT(_exp) _Static_assert(_exp, "static assert failed")
 
+#define STR0(x) #x
+#define STR(x) STR0(x)
+
+#define UF2_ARRAY_SIZE(_arr)    ( sizeof(_arr) / sizeof(_arr[0]) )
```

Comment:
This is not a typesafe macro, and has historically been the source of many really troublesome bugs.
I strongly recommend using a typesafe alternative.

<details><summary>Recommended alternative definition</summary><P/>

```C++

#if defined(__cplusplus)
     /*
        Works on older compilers, even Visual C++ 6....
        Created by Ivan J. Johnson, March 06, 2007
        See http://drdobbs.com/cpp/197800525?pgno=1
        Full description is in markdown file array_size2.md
    */
    #define ARRAY_SIZE2(arr) ( \
       0 * sizeof(reinterpret_cast<const ::Bad_arg_to_ARRAY_SIZE2*>(arr)) + /*check1*/ \
       0 * sizeof(::Bad_arg_to_ARRAY_SIZE2::check_type((arr), &(arr)))    + /*check2*/ \
       sizeof(arr) / sizeof((arr)[0])                                       /* eval */ \
       )

    struct Bad_arg_to_ARRAY_SIZE2 {
       class Is_pointer; // incomplete
       class Is_array {};
       template <typename T>
       static Is_pointer check_type(T const *, T const * const *);
       static Is_array check_type(void const *, void const *);
    };
#elif !defined(__cplusplus) && defined(__GNUC__)

    // Even C can have type-safety for equivalent of ARRAY_SIZE() macro,
    // when using the following two GCC extensions:
    //     typeof()
    //     __builtin_types_compatible_p()

    #if defined(ARRAYSIZE2_SHOW_VERSION_MESSAGE)
        #pragma message( "ARRAY_SIZE2 -- Using GNUC version" )
    #endif

    // validated using:
    //   MSP430  gcc   4.5.3
    //   x86-64  icc  16.0.3
    //   x86-64  gcc   4.1.2
    //   x86-64 clang  3.0.0
    //   AVR     gcc   4.5.4
    //   ARM     gcc   4.5.4

    #define __SIMPLEHACKS_COMPATIBLE_TYPES__(a,b)   __builtin_types_compatible_p(typeof(a), typeof(b)) // GCC extensions
    #define __SIMPLEHACKS_BUILD_ERROR_ON_NONZERO__(x)  (sizeof(struct { int:-!!(x)*0x1ee7;})) // if x is zero, reports "error: negative width in bit-field '<anonymous>'"
    #define __SIMPLEHACKS_MUST_BE_ARRAY__(x)        __SIMPLEHACKS_BUILD_ERROR_ON_NONZERO__(__SIMPLEHACKS_COMPATIBLE_TYPES__((x), &(*x)))
    #define ARRAY_SIZE2(_arr)       ( (sizeof(_arr) / sizeof((_arr)[0])) + __SIMPLEHACKS_MUST_BE_ARRAY__(_arr) ) // compile-time error if not an array

```

</details><HR/>

Better yet, just use the header from https://github.com/SimpleHacks/UtilHeaders/ (MIT Licensed).

Reply:
Yeah, I am aware of this. It is only dangerous when using with dynamic pointer/reference. Since we only use it for getting the number of static files locally. It is rather safe, though I will update it to declare NUM_FILES as enum later on.